pypeit/core/extract.py

```diff
@@ -175,10 +175,16 @@ def extract_optimal(sciimg, ivar, mask, waveimg, skyimg, thismask, oprof,
                                    np.nansum(thismask_sub[oprof_good,:]*oprof_sub[oprof_good,:]**2, axis=1)
         oprof_bad = badwvs & ((oprof_smash <= 0.0) | (np.isfinite(oprof_smash) == False) | (wave_opt <= 0.0) | (np.isfinite(wave_opt) == False))
         if oprof_bad.any():
+            # If there are no good profile wavelengths, use boxcar wavelengths for these pixels
+            # get boxcar_radius
+            box_radius = spec.BOX_RADIUS
```

Comment:
Can you explain why the change is needed and effective?
  Thx.

Reply:
The interpolation is problematic if lots of pixels get masked. The current fix is more robust.